### PR TITLE
1508 json extension

### PR DESCRIPTION
**Pull request by captDaylight**
Allow for json to be imported without extension.

ref #1508

### CODE REVIEW

**Comment by lawrencetvo** [comment]
@captDaylight commenting here because it makes more sense to be honest. Your coverage is failing because the commit with the description `remove jsx` is failing. 

It seems that you removed the `.jsx` extension from the loaders.

**Comment by captDaylight** [comment]
@lawrencetvo yup commenting here is better :) @arunoda said to remove .jsx from the resolved extension list, I don't think that is the problem. Looks like it has to do with: 

`npm ERR! Failed at the next@2.0.0 testonly script 'cross-env NODE_PATH=test/lib jest \.test.js "--coverage" "--forceExit" "--runInBand" "--verbose" "--bail"'.`

**Comment by lawrencetvo** [comment]
@captDaylight I'll check out your branch later.

**Comment by lawrencetvo** [comment]
@captDaylight what script are you running?

**Comment by lawrencetvo** [comment]
@captDaylight this might be noobish on my end but the browser doesn't seem to be opening on a port. It keeps coming out as `localhost:undefined`

**Comment by lawrencetvo** [comment]
On a side note, where is the `next-test-utils` repo?

File: server/build/webpack.js [hunk 1]
```diff
@@ -280,6 +280,7 @@ export default async function createCompiler (dir, { dev = false, quiet = false,
       }
     },
     resolve: {
+      extensions: ['', '.js', '.json'],
```
Comment (arunoda) [inline server/build/webpack.js]: I think we can't have empty string here.
That's the root cause for this issue: https://travis-ci.org/zeit/next.js/jobs/216011309

**Comment by arunoda** [comment]
And write a [test case](https://github.com/zeit/next.js/tree/master/test/integration/basic) here with this usage.

Basically, you can create a page import a .json file which gives you some data.
Render that in the page and check the content via the test.

**Comment by lawrencetvo** [comment]
@arunoda is there not a test file already in `~/test/pages/json.js`?

**Comment by arunoda** [comment]
@lawrencetvo yep there's. I need to look at why we need this in that case.

**Comment by arunoda** [comment]
@captDaylight thanks for the PR.
But with webpack2 json works by default.

**Pull request closed without merging**